### ISSUE TITLE
hyperlane-deploy/sdk errors should be more accessible 

### ISSUE THREAD

**Issue by nambrot** [hyperlane-xyz/hyperlane-monorepo#2158]
From: https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/2115

- Fallback/retry provider aggregates the underlying errors which makes them hard to parse and produce large logs
- Most people will only specify one provider, thus its the wrapping of the retry provider which presumably create most of the problems

Possible solution:
- We should modify retry provider to catch and identify the actual underlying error for better legibility

**Comment by avious00** [comment]
@jmrossy should we close this now that we have smartprovider in sdk?

**Comment by jmrossy** [comment]
@avious00 yes indeed. Closing
